pandas/io/json.py

```diff
@@ -884,3 +939,174 @@ def _recursive_extract(data, path, seen_meta, level=0):
         result[k] = np.array(v).repeat(lengths)
 
     return result
+
+
+# ---------------------------------------------------------------------
+# JSON-Table Schema routines
+# http://specs.frictionlessdata.io/json-table-schema/
+
+
+def as_json_table_type(x):
```

Comment:
I would make these set of functions into a class. *maybe* put in another module, ``table_schema``?